File: src/test/java/com/raychenon/leetcode/CustomSortStringTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-11-06
 * https://leetcode.com/problems/custom-sort-string/
 */
public class CustomSortStringTest {

    private static CustomSortString instance;

    @BeforeClass
    public static void setUp() {
        instance = new CustomSortString();
    }

    @Test
    public void customSort_cbad_Test() {
        assertEquals("cbad", instance.customSortString("cba", "abcd"));
    }

    @Test
    public void customSort_cbaaxxyyzz_Test() {
        assertEquals("cbaaxxyyzz", instance.customSortString("cba", "bxzyacaxyz"));
    }
}
                    

File: src/test/kotlin/com/raychenon/kotlin/leetcode/CustomSortStringTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase.assertEquals
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-11-06
 * https://leetcode.com/problems/custom-sort-string/
 */
class CustomSortStringTest {

    @Test
    fun sort_cbad_Test() {
        assertEquals("cbad", CustomSortString.customSortString("cba", "abcd"))
    }

    @Test
    fun sort_cbaaxxzzyy_Test() {
        assertEquals("cbaaxxzzyy", CustomSortString.customSortString("cba", "bxzyacaxyz"))
    }

}                                                                        